src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -303,9 +303,11 @@ protected function normalizeValue($value)
             if (isset($this->children[$name])) {
                 $normalized[$name] = $this->children[$name]->normalize($val);
                 unset($value[$name]);
-            } elseif (false === $this->removeExtraKeys) {
+            } elseif (!$this->removeExtraKeys) {
                 $normalized[$name] = $val;
-                unset($value[$name]);
+                if ($this->ignoreExtraKeys) {
+                    unset($value[$name]);
```

Comment:
When thinking once more about it, I think the `unset` along with the condition is unnecessary, as the `InvalidConfigurationException` is thrown only when not ignoring extra keys. Removing the key don't have an effect on the method result. As an evidence, removing these three lines don't makes the unit tests fail. What do you think?